docs/rfcs/003-consolidate-core-packages.md

```diff
@@ -0,0 +1,338 @@
+
+On the other hand, having these packages live in separate repositories and released "independently" introduces a great deal of overhead when adding new features.  Here is a comparison of the current package development workflow contrasted to what we could achieve with consolidated packages:
+
+#### Current Package Development Workflow
+
+For example, to add a single feature to the `tree-view` package, one must:
+
+1. Fork and clone the `tree-view` repository to their computer (making sure to pull the commit relevant to the version of Atom they are working with)
+1. Run `apm install` and `apm link` inside of the repo folder
+1. Make their desired changes to the code
+1. Open a PR to the `tree-view` repo and wait for CI to pass and a maintainer to review it
+1. Work with maintainers to get the PR approved and merged
+
+After this is finished, an Atom maintainer must take the following steps:
+
+1. Clone the `tree-view` repo
+2. Run `apm publish` to publish a new release of the package
+3. Edit `package.json` in the Atom repo to reflect the new version of `tree-view`
+4. Commit and push the changes to the relevant branch where the change belongs (`master` or `1.nn-releases`)
+
+#### Simplified Package Development
+
+If we were to move `tree-view` (or any other core Atom package) back into `atom/atom`, the development workflow would look more like this:
+
+1. Fork and clone `atom/atom` and switch to a release branch if necessary
+1. Build Atom and launch it in dev mode
+1. Make desired changes to the code in `packages/tree-view`
+1. Open a PR on `atom/atom` and wait for CI to pass and a maintainer to review it
+1. Work with maintainers to get the PR approved and merged
+
+At this point, the change is merged into Atom and ready for inclusion in the next release.
+
+### Greatly reduced burden for maintainers
+
+Since packages all have their own repositories, this means that we have to watch 91 different repos for issues and pull requests.  This also means that we have to redirect issues filed on `atom/atom` to the appropriate repository when a user doesn't know where it belongs.  Even more importantly, there's not an easy way to prioritize and track issues across the Atom organization without using GitHub Projects.
+
+Also, as mentioned above, there's the added duty of doing the package "version dance" when we merge any new PRs to a package repository: publish the package update, update `package.json` in Atom.  It's very easy to forget to do this and not have community contributions included in the next Atom release!
+
+The more core packages live in `atom/atom`, the less work Atom maintainers have to do overall.
+
+## Explanation
+
+Many of Atom's core packages now live in the core `atom/atom` repository.  To the Atom user, this change will be imperceptible as these packages still show up in the list of Core Packages in the Settings View.  Users can still disable these packages if they want to replace their behavior with other packages.
```

Comment:
> Users can still disable these packages if they want to replace their behavior with other packages.

You could disable a package with no intention of replacing its functionality. Right?

Suggested edit:

> Users can still optionally disable these packages.

Reply:
Yep!  I'll make that change.